spec/requests/users_controller_spec.rb

```diff
@@ -3071,6 +3071,54 @@ def post_user(extra_params = {})
         expect(response.status).to eq(200)
       end
     end
+
+    context 'avatar flair in Most... sections' do
+      it "returns data for automatic groups flair" do
+        liker = Fabricate(:user, admin: true, moderator: true, trust_level: 1)
+        create_and_like_post(user, liker)
+
+        get "/u/#{user.username_lower}/summary.json"
+        json = response.parsed_body
+
+        expect(json["user_summary"]["most_liked_by_users"][0]["admin"]).to eq(true)
+        expect(json["user_summary"]["most_liked_by_users"][0]["moderator"]).to eq(true)
+        expect(json["user_summary"]["most_liked_by_users"][0]["trust_level"]).to eq(1)
```

Comment:
What does it add to test that? Aren't these fields already being tested "implicitly" already?

Reply:
We just need to be sure that the server returns it to the client-side component. It wasn't working because we didn't return the right information from the server. The component on the client-side is on the place and in the order.

I think it isn't a very good separation of logic between client and server and a pretty tricky thing to test.